dali/test/python/decoder/test_imgcodec.py

```diff
@@ -404,3 +406,36 @@ def test_peek_shape():
         'tiff/0/kitty-2948404_640.tiff', (433, 640, 1), types.GRAY, True
     yield _testimpl_image_decoder_peek_shape, \
         'bmp/0/cat-111793_640_grayscale.bmp', (426, 640, 3), types.RGB, True
+
+
+@params(
+        ('cat-1245673_640_grayscale_16bit', types.ANY_DATA, types.UINT16),
+        ('cat-3449999_640_grayscale_16bit', types.ANY_DATA, types.UINT16),
+        ('cat-3449999_640_grayscale_12bit', types.ANY_DATA, types.UINT16),
+        ('cat-3449999_640_grayscale_16bit', types.GRAY, types.UINT16),
+        ('cat-3449999_640_grayscale_8bit', types.ANY_DATA, types.UINT8),
+)
+def test_image_decoder_lossless_jpeg(img_name, output_type, dtype):
+    data_dir = os.path.join(test_data_root, "db/single/jpeg_lossless/0")
+    ref_data_dir = os.path.join(test_data_root, "db/single/reference/jpeg_lossless")
+
+    @pipeline_def(batch_size=1, device_id=0, num_threads=1)
+    def pipe(file):
+        encoded, _ = fn.readers.file(files=[file])
+        decoded = fn.experimental.decoders.image(
+                encoded, device='mixed', dtype=dtype, output_type=output_type)
+        return decoded
+
+    def run(file):
+        p = pipe(file)
+        p.build()
+        out, = p.run()
+        return np.array(out[0].as_cpu())
+
+    result = run(data_dir + f'/{img_name}.jpg')
+    ref = np.load(ref_data_dir + f'/{img_name}.npy')
+    kwargs = {}
+    # account for rounding errors when comparing to the reference (generated)
+    if '12bit' in img_name:
+        kwargs['atol'] = 1
```

Comment:
When we switch to an FP reference, this can be `0.5 + eps` (for 12-bit 0.501 would admit 2 ULP error).